alchemiscale/storage/statestore.py

```diff
@@ -831,33 +967,51 @@ def dequeue_taskqueue_tasks(
         """
         for t in tasks:
             q = f"""
-            // get our task queue, as well as the task we want to remove, and
-            // the nodes ahead and behind it 
-            MATCH (task:Task {{_scoped_key: '{t}'}}),
-                  (behind)-[behindf:FOLLOWS {{taskqueue: '{taskqueue}'}}]->(task),
-                  (task)-[aheadf:FOLLOWS {{taskqueue: '{taskqueue}'}}]->(ahead)
-            WITH behind, behindf, task, aheadf, ahead
-
-            // create connection between behind and ahead nodes
-            CREATE (behind)-[newf:FOLLOWS {{taskqueue: '{taskqueue}'}}]->(ahead)
-
-            // delete connections between node to remove and behind, ahead nodes
-            DELETE behindf, aheadf
-
+            // get our task hub, as well as the task we want to remove
+            MATCH (th:TaskHub {{_scoped_key: '{taskhub}'}})-[:ACTIONS]->(task:Task {{_scoped_key: '{t}'}}),
+            DETACH DELETE task
```

Comment:
We don't want to actually delete the `Task` here; `cancel_tasks` should be the opposite of `action_tasks`, so it should remove the `ACTIONS` relationship between the given `Task`s and the `TaskHub`.

Reply:
Sorted.